frontend/appflowy_flutter/lib/shared/icon_emoji_picker/recent_icons.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:convert';
+
+import 'package:appflowy/shared/icon_emoji_picker/icon.dart';
+import 'package:appflowy_backend/log.dart';
+import 'package:flutter/foundation.dart';
+
+import '../../core/config/kv.dart';
+import '../../core/config/kv_keys.dart';
+import '../../startup/startup.dart';
+import 'flowy_icon_emoji_picker.dart';
+
+class RecentIcons {
+  static final Map<String, List<String>> _dataMap = {};
+  static bool _loaded = false;
+  static const maxLength = 20;
+
+  ///prevent the Recent Icon feature from affecting the unit tests of the Icon Selector.
+  @visibleForTesting
+  static bool enable = true;
+
+  static Future putEmoji(String id) async {
+    await _put(FlowyIconType.emoji, id);
+  }
+
+  static Future putIcon(Icon icon) async {
+    await _put(
+      FlowyIconType.icon,
+      jsonEncode(
+        Icon(name: icon.name, keywords: icon.keywords, content: icon.content)
+            .toJson(),
+      ),
+    );
+  }
+
+  static Future<List<String>> getEmojiIds() async {
+    await _load();
+    return _dataMap[FlowyIconType.emoji.name] ?? [];
+  }
+
+  static Future<List<Icon>> getIcons() async {
+    await _load();
+    final iconList = _dataMap[FlowyIconType.icon.name] ?? [];
+    try {
+      return iconList
+          .map((e) => Icon.fromJson(jsonDecode(e) as Map<String, dynamic>))
+          .toList();
+    } on FormatException catch (e) {
+      Log.error('RecentIcons getIcons with :$iconList', e);
+    } on TypeError catch (e) {
+      Log.error('RecentIcons getIcons with :$iconList', e);
+    }
+    return [];
+  }
+
+  static Future<void> _save() async {
+    await getIt<KeyValueStorage>().set(
+      KVKeys.kRecentIcons,
+      jsonEncode(_dataMap),
+    );
+  }
+
+  static Future _load() async {
```

Comment:
Still missing return type on methods in this class.

```suggestion
  static Future<void> _load() async {
```